chordspy/connection.py

```
@@ -0,0 +1,681 @@
+        if self.recording_active:
+            return False
+        
+        try:
+            # Generate filename if not provided
+            if not filename:
+                timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+                filename = f"ChordsPy_{timestamp}.csv"
+            elif not filename.endswith('.csv'):
+                filename += '.csv'
+                
+            self.csv_file = open(filename, 'w', newline='')         # Open CSV file and initialize writer
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)